server/app/models/stack_revision.rb

```diff
@@ -12,10 +12,12 @@ class StackRevision
   field :expose, type: String
   field :services, type: Array
   field :volumes, type: Array
+  field :labels, type: Array, default: []
 
   belongs_to :stack
 
   index({ stack_id: 1 })
+  index({ labels: 1 })
```

Comment:
Not very sure about indexing `labels`... I used `host_node` as a primer to derive from and my assumptions might be incompetent at best. 😃